docs/src/main/asciidoc/spring-cloud-sleuth.adoc

```diff
@@ -60,6 +57,123 @@ NOTE: Remember that tags are only collected and exported if there is a
+assertThat(continuedSpan).isEqualTo(spanToContinue);
+----
+- *detach* - the span doesn't get stopped or closed. It only gets removed from the current thread.
+- *join* - you can create a new span and set an explicit parent to it
+
+=== Creating spans
+
+You can manually create spans by using the *Tracer* interface.
+
+[source,java]
+----
+include::../../../../spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java[tags=manual_span_creation,indent=0]
+----
+
+In this example we could see how to create a new instance of span. Assuming that there already
+was a span present in this thread then it would become the parent of that span.
+
+=== Continuing spans
+
+Sometimes you don't want to create a new span but you want to continue one. Example of such a
+situation might be (of course it all depends on the use-case):
+    - *AOP* - If there was already a span created before an aspect was reached then you might not want to create a new span.
+    - *Hystrix* - executing a Hystrix command is most likely a logical part of the current processing. It's in fact
+    only a technical implementation detail that you wouldn't necessarily want to reflect in tracing as a separate being.
+
+To continue a span you can use the *Tracer* interface.
+
+[source,java]
+----
+include::../../../../spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java[tags=manual_span_continuation,indent=0]
+----
+
+What's important is to always clean after you create a span. The golden rule would be to:
+ - close a span if you want to send it to Zipkin
+ - detach a span if some work was done started in one thread (e.g. thread X) and it's waiting for other threads (e.g. Y, Z)
+ to finish. Then the spans in the threads Y, Z should be detached at the end of their work. When the results are collected
+ the span in thread X should be closed.
+
+=== Joining spans
+
+There is a possibility that you want to start a new span and provide an explicit parent of that span.
+Let's assume that the parent of a span is in one thread and you want to start a new span in another thread. The
+`joinTrace` method of the `Tracer` interface is the method you are looking for.
+
+[source,java]
+----
+include::../../../../spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/documentation/SpringCloudSleuthDocTests.java[tags=manual_span_joining,indent=0]
+----
+
+IMPORTANT: After having joined the span remember to close it. Otherwise you will see a lot of warnings in your logs
+ related to the fact that you have a span present in the current thread other than the one you're trying to close.
+ What's worse your spans won't get closed properly thus will not get collected to Zipkin.
+
+== Naming spans
+
+Picking a span name is not a trivial task. Span name should depict an operation
+but shouldn't be too long (you don't want the Zipkin UI to render for minutes,
```

Comment:
this isn't about length, this is about cardinality. operation names should be low cardinality, ex not include identifiers.